packages/blade/src/utils/TooltipifyComponent.tsx

```diff
@@ -0,0 +1,43 @@
+import type { TooltipProps } from '~components/Tooltip';
+import { Tooltip } from '~components/Tooltip';
+
+type TooltipifyComponentProps = {
+  children: React.ReactElement;
+  tooltip?: Pick<TooltipProps, 'title' | 'content' | 'onOpenChange'>;
+};
+
+/**
+ * Utility to support tooltip as a prop
+ *
+ * Example Usage:
+ *
+ * ```jsx
+ * const SideNavLink = ({ tooltip }) => {
+ *   return (
+ *    <TooltipifyComponent tooltip={tooltip}>
+ *     <button>Click me</button>
+ *    </TooltipifyComponent>
+ *   )
+ *
+ * }
+ * ```
+ *
+ * This adds a tooltip to the button when `tooltip` prop is passed. Otherwise renders the children as is
+ */
+const TooltipifyComponent = ({
```

Comment:
Is this exposed to consumers?

Reply:
nope internal util. Using it in SideNavLink and we might need this if we decide to add tooltip to ActionListItem